Sources/protoc-gen-swift/ProtoNameInstructionWriter.swift

```diff
@@ -0,0 +1,94 @@
+// Sources/protoc-gen-swift/ProtoNameInstructionWriter.swift - Name instruction writing helpers
+//
+// Copyright (c) 2014 - 2025 Apple Inc. and the project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See LICENSE.txt for license information:
+// https://github.com/apple/swift-protobuf/blob/main/LICENSE.txt
+//
+// -----------------------------------------------------------------------------
+
+import SwiftProtobuf
+import SwiftProtobufPluginLibrary
+
+/// A convenience wrapper that adds operations to write `ProtoNameInstruction`s and their operands
+/// into bytecode streams.
+struct ProtoNameInstructionWriter {
+    private(set) var bytecode = BytecodeWriter<ProtoNameInstruction>()
+
+    /// The previous field or case number written to the stream, which is used to compute deltas.
+    private var previousNumber: Int32 = 0
+
+    mutating func writeSame(number: Int32, name: String) {
+        let delta = number - previousNumber
+        previousNumber = number
```

Comment:
If you want a safety check for the sorta case I flagged, I believe with what we're doing now, we could put asserts in there to chat if the delta ever went negative.

Reply:
Done.